src/storage/src/mut_records.rs

```diff
@@ -104,11 +114,16 @@ impl MutFileRecords {
     }
 
     pub async fn validate(&mut self) -> Result<Offset, LogValidationError> {
-        validate(self.f_sink.get_path()).await
+        let f_sink = self.f_sink.lock().await;
+        validate(f_sink.get_path()).await
     }
 
     pub fn get_pos(&self) -> Size {
-        self.f_sink.get_current_len() as Size
+        block_on(async {
```

Comment:
I would rather not see block_on here.  There maybe two path I see:
1) change this to async
2) cache len

Reply:
yup can do there, I'll explore both for this

Reply:
Returning cached get_pos value now.  One thing that makes me uneasy working with the change is that the position is a u64 and the Size type is u32; this existed before any of of these changes.  But that means there might be some funny behavior showing up at 4GB log file sizes.  cargo clippy can turn on lints for casting truncations, but this might want to be a separate issue.

Reply:
can mitigate by only allowing limit segment up to u32.  Typical size for segment is 1G

Reply:
I'm thinking it might be worth putting in a front end check for overflow of 4GB (or 1GB) on write.